actions.yaml

```diff
@@ -63,7 +63,9 @@ user-create:
   params:
     name:
       type: string
-      description: Username for the new user
+      description: |
+        Username for the new user. This value must only contain alphanumeric
+        characters, '@', '-' or '.'.
```

Comment:
Should `:` be allowed, too? I ask because `:` is a prominent character used in a lot of user names, as seen from `user-list`:

```
$ juju run-action --wait kubernetes-master/0 user-list
unit-kubernetes-master-0:
  UnitId: kubernetes-master/0
  id: "16"
  results:
    users: admin, Bob, bob@foo, bob, system:kube-controller-manager, system:kube-proxy,
      system:node:ip-172-31-47-88, system:node:ip-172-31-8-169, system:node:ip-172-31-21-142,
      system:kube-scheduler, system:monitoring
  status: completed
  timing:
    completed: 2021-01-15 20:08:43 +0000 UTC
    enqueued: 2021-01-15 20:08:41 +0000 UTC
    started: 2021-01-15 20:08:42 +0000 UTC
```

which, as a dumb user, makes me think I might be able to create a user named `custom:bob`, but no:

```
$ juju run-action --wait kubernetes-master/0 user-create name=custom:bob
unit-kubernetes-master-0:
  UnitId: kubernetes-master/0
  id: "20"
  message: User name may only contain alphanumeric characters, '@', '-' or '.'
  results:
    users: admin, Bob, bob@foo, bob, system:kube-controller-manager, system:kube-proxy,
      system:node:ip-172-31-47-88, system:node:ip-172-31-8-169, system:node:ip-172-31-21-142,
      system:kube-scheduler, system:monitoring
  status: failed
  timing:
    completed: 2021-01-15 20:12:14 +0000 UTC
    enqueued: 2021-01-15 20:12:08 +0000 UTC
    started: 2021-01-15 20:12:13 +0000 UTC
```

Reply:
I think it could work to allow `:` if we updated [auth-webhook's delimiter handling](https://github.com/charmed-kubernetes/charm-kubernetes-master/blob/bb3328c490f2e05b20010f688be55296623fb7e1/templates/cdk.master.auth-webhook.py#L148) to split on only the right-most occurrence of `::`.

I don't think this is a show stopper though.